__main__.py

```diff
@@ -0,0 +1,58 @@
+"""
+This is the main entry point for SlateDB's Pulumi configuration. It manages
+memberships and permissions for SlateDB's open source resources such as
+Github, Discord, and Crates.io.
+"""
+
+from dataclasses import dataclass
+
+from resources import crates, discord, github
+
+@dataclass
+class Maintainer:
+    name: str
+    github: str
+    discord: str
+
+
+MAINTAINERS = [
+    Maintainer(
+        name="Chris Riccomini",
+        github="criccomini",
+        discord="839536195603922974",
+    ),
+    # Maintainer(
+    #     name="Rohan Desai",
+    #     github="rodesai",
+    #     discord="1144686022010097794",
+    # ),
+    # Maintainer(
+    #     name="Vignesh Chandramohan",
+    #     github="vigneshc",
+    #     discord="894942567513337887",
+    # ),
+    # Maintainer(
+    #     name="Paul Butler",
+    #     github="paulgb",
+    #     discord="",
+    # ),
+    # Maintainer(
+    #     name="Li Yazhou",
+    #     github="flaneur2020",
+    #     discord="965845316370833408",
+    # ),
+]
+
+
+def main():
```

Comment:
Is this meant to be run whenever there are new steps or new maintainers? Are they expected to be idempotent?

Reply:
i guess it should be idempotent, pulumi's programming model is somewhat looks like imperative but actually declarative. like i wanna have a database, i'd call `create_db()`, and pulumi should track the added databases and other allocated cloud resources in a state file, then decide what to add/remove by making a diff with the earlier state file.

Reply:
I see. What happens if some config changes outside of pulumi? For example, `create_db()` was called, but someone manually deleted the DB afterwards.

Reply:
IaC mostly assumes all the related resources are managed in the IaC code, besides the identifiers like "mydb01", the resource id of this database in AWS is also tracked in the state file. if the ids conflicts between the state file & the real cloud resources, it seems that pulumi tends to raise an error for us, and let us resolve it manually or fix the state file manually, it'd be a pain 🥲

Reply:
Yep, Pulumi only manages what it knows about. It leaves other stuff alone. In cases where there's a conflict, it just fails and lets the user fix things.

In our case, I'm planning to run this for our committer access management on various resources. Right now, We already have owners defined for Github and Crates. For Discord, committers are already added to the committers role. I checked Github and Pulumi seems to just update stuff if it already exists, so shouldn't cause an issue. For Crates and Discord, we'll have to see--this is one of the reasons why I just added myself. 😅 

Once this is deployed, I plan to go and remove all other access to the Github org. I added a ton of people as members with write access to the org or various repos. I am going to remove all of that.